content/03-reference/01-tools-and-interfaces/02-prisma-client/02-constructor.mdx

```diff
@@ -0,0 +1,70 @@
+---
+title: 'Constructor'
+metaTitle: 'Prisma Client Constructor (Reference)'
+metaDescription: 'This page describes the Prisma Client constructor parameters.'
+---
+
+## Overview
+
+This page describes the Prisma Client constructor parameters.
+
+## `datasources`
+
+Use the `datasources` constructor parameter to programatically override the URL of a datasource - for example, as part of an integration test. To override a datasource:
+
+1. Define a datasource in the `schema.prisma` file. In the following example, there is a single datasource named `my_database`:
+
+   ```prisma
+   datasource my_database {
+     provider = "sqlite"
+     url      = env("DATABASE_URL")
+   }
+   ```
+   
+   The `DATABASE_URL` environment variable defines the following value:
+
+   ```
+   DATABASE_URL="file:./dev.db"
+   ```
+
+   > **Note**: You must re-generate the Prisma Client each time you add or rename a datasource. Datasource names are included in the generated client.
+
+2. Instantiate a client and override the connection string of the `my_database` datasource:
+
+   ```ts
+   import { PrismaClient } from '@prisma/client'
+
+   const prisma = new PrismaClient({
+     datasources: {
+   |   my_database: 'file:./dev_qa.db',
+     },
+   })
+   ```
+
+> **Note**: If you define multiple datasources in `schema.prisma`, Prisma defaults to the first entry.
```

Comment:
Why is that? Do we not define which one it should be applied to by using `my_database`?

Reply:
cc @timsuchanek

Reply:
There was a enabled property in the schema before that got removed (never saw it, just the remains)

Reply:
What is the connection to this comment? Should it not use the datasource with the name you supply above, `my_database` in this example?

Reply:
This is actually touching on undefined behavior, I think the **Note** is correct, but we need to fix that.
Even though you provide a specific datasource as a constructor arg, it could even be, that in our current implementation, it might not be used, if it's not the first datasource in the schema.
We don't have tests for that and didn't define this behavior clearly yet, but should definitely do that asap.